compiler/src/dotty/tools/dotc/transform/IsInstanceOfEvaluator.scala

```diff
@@ -126,6 +127,25 @@ class IsInstanceOfEvaluator extends MiniPhaseTransform { thisTransformer =>
 
           val selClassNonFinal = selClass && !(selector.typeSymbol is Final)
           val selFinalClass    = selClass && (selector.typeSymbol is Final)
+          val selTypeParam     = tree.args.head.tpe.widen match {
+            case AppliedType(tycon, args) =>
+              ctx.uncheckedWarning(
+                ErasedType(hl"""|Since type parameters are erased, you should not match on them in
```

Comment:
There's one exception to this, and that's arrays.

    case xs: Array[Int]

_is_ testable. We should (a) check that it is indeed tested and (b) adapt the error condition.